e2e-tests/smart-update3/run

```diff
@@ -0,0 +1,214 @@
+
+	##################################################
+	desc 'PMM2 cluster update with the recommended image by version service'
+
+	# Prepare cluster config
+	cp -f "${test_dir}/conf/${CLUSTER}-version-service-unreachable.yml" "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+	cat "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+	yq -i eval ".spec.initContainer.image = \"${IMAGE}\"" "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+
+	desc "DEBUG"
+	cat "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+	desc "DEBUG"
+	spinup_pxc "${CLUSTER}" "${tmp_dir}/${CLUSTER}-version-service-unreachable.yml"
+
+
+	# Get actual PXC version from running cluster and add to version service
+	wait_cluster_consistency "${CLUSTER}" "${CLUSTER_SIZE}" "${PROXY_SIZE}"
+	ACTUAL_PXC_VERSION=$(kubectl_bin get pxc "${CLUSTER}" -o jsonpath='{.status.pxc.version}')
+	desc "Detected PXC version: ${ACTUAL_PXC_VERSION}"
+	add_pxc_version_to_vs "${ACTUAL_PXC_VERSION}" "${IMAGE_PXC}"
+
+	# Enable PMM2 with older version
+	kubectl_bin patch pxc/"${CLUSTER}" --type=merge -p '{"spec":{"pmm":{"enabled":true,"image":"percona/pmm-client:2.44.0","serverHost":"monitoring-service"}}}'
+	wait_cluster_consistency "${CLUSTER}" "${CLUSTER_SIZE}" "${PROXY_SIZE}"
+
+	# Get initial PMM2 image
+	initial_pmm2_image=$(kubectl_bin get pod "${CLUSTER}-pxc-0" -o jsonpath='{.status.containerStatuses[?(@.name=="pmm-client")].image}')
+	desc "Initial PMM2 image: ${initial_pmm2_image}"
+
+	# Get recommended PMM2 image from version service
+	pmm2_recommended_image=$(kubectl_bin exec -ti "$(get_operator_pod)" ${OPERATOR_NS:+-n $OPERATOR_NS} -- curl -s "${VS_URL}.${namespace}.svc.cluster.local:${VS_PORT}/versions/v1/pxc-operator/9.9.9" | jq -r '.versions[].matrix.pmm | to_entries[] | select(.value.imagePath) | select(.value.imagePath | contains("pmm-client:2")) | select(.value.status == "recommended") | .value.imagePath')
+	desc "Target PMM2 recommended image: ${pmm2_recommended_image}"
+
+	# Update cluster to use version service with short schedule
+	kubectl_bin patch pxc/"${CLUSTER}" --type=merge -p '{"spec":{"upgradeOptions":{"versionServiceEndpoint":"'${VS_ENDPOINT}'","apply":"recommended","schedule": "* * * * *"}}}'
+
+	# Wait for PMM2 update
+	wait_pmm_update "${CLUSTER}" "${CLUSTER_SIZE}" "${pmm2_recommended_image}" "2" || exit 1
+
+	wait_cluster_consistency "${CLUSTER}" "${CLUSTER_SIZE}" "${PROXY_SIZE}"
+
+	# Verify PMM2 updated
+	for i in $(seq 0 $((CLUSTER_SIZE - 1))); do
+		actual_pmm_image=$(kubectl_bin get pod "${CLUSTER}-pxc-${i}" -o jsonpath='{.status.containerStatuses[?(@.name=="pmm-client")].image}')
+		if [[ "${actual_pmm_image}" != *"${pmm2_recommended_image}"* ]]; then
+			echo "ERROR: PMM2 image not updated on ${CLUSTER}-pxc-${i}. Expected: ${pmm2_recommended_image}, Got: ${actual_pmm_image}"
+			exit 1
+		fi
+	done
+	desc "PMM2 successfully updated to ${pmm2_recommended_image}"
```

Comment:
Can we add a check that pmm works? For emample fetch some data from PMM after the update?